### PR TITLE
Add rocBLAS_jll and rocSPARSE_jll deps

### PR DESCRIPTION
**Pull request by jpsamaroo**
And upper-bound all ROCm JLLs to 5 in preparation for upcoming ROCm 5.x JLLs

@pxl-th

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -378,39 +384,31 @@ function main()
 
         if !use_artifacts
```
Comment (pxl-th) [inline deps/build.jl:385]: I think this is not needed. As it will never hit `use_artufacts` case below.

Comment (jpsamaroo) [reply deps/build.jl]: Thanks for catching this!

File: deps/build.jl [hunk 2]
```diff
@@ -376,41 +382,31 @@ function main()
...
         config[:hip_configured] = true
 
-        if !use_artifacts
-            ### Find external HIP-based libraries
-            for name in ("rocblas", "rocsolver", "rocsparse", "rocalution", "rocfft", "MIOpen")
-                lib = Symbol("lib$(lowercase(name))")
-                path = find_rocm_library("lib$name", roc_dirs)
+        ### Find external HIP-based libraries
+        for (name, pkg) in rocm_ext_libs
+            lib = Symbol("lib$(lowercase(name))")
```
Comment (pxl-th) [inline deps/build.jl:387]:
```suggestion
            llib = Symbol(:lib, lowercase(string(name)))
```

File: deps/build.jl [hunk 3]
```diff
@@ -376,41 +382,31 @@ function main()
...
+            if use_artifacts && pkg !== nothing
+                try
+                    @eval using $pkg
+                    config[lib] = Symbol(pkg, :., lib)
```
Comment (pxl-th) [inline deps/build.jl:391]:
```suggestion
                    config[lib] = getfield(@eval($pkg), lib)
```

**Pull request merged**